governance/program/src/state/enums.rs

```diff
@@ -92,3 +92,60 @@ impl Default for ProposalState {
         ProposalState::Draft
     }
 }
+
+/// The type of the vote threshold percentage used to resolve a vote on a Proposal
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum VoteThresholdPercentageType {
+    /// Voting threshold of Yes votes in % required to tip the vote
+    /// It's the percentage of tokens out of the entire pool of governance tokens eligible to vote
+    /// Note: If the threshold is below or equal to 50% then an even split of votes ex: 50:50 or 40:40 is always resolved as Defeated
+    /// In other words +1 vote tie breaker is always required to have a successful vote
+    YesVote,
+
+    /// The minimum number of votes in % out of the entire pool of governance tokens eligible to vote
+    /// which must be cast for the vote to be valid
+    /// Once the quorum is achieved a simple majority (50%+1) of Yes votes is required for the vote to succeed
+    /// Note: Quorum is not implemented in the current version
+    Quorum,
+}
+
+/// The source of voter weights used to vote on proposals
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum VoteWeightSource {
+    /// Governing token deposits into the Realm are used as voter weights
+    Deposit,
+    /// Governing token account snapshots as of the time a proposal entered voting state are used as voter weights
+    /// Note: Snapshot source is not supported in the current version
+    /// Support for account snapshots are required in solana and/or arweave as a prerequisite
+    Snapshot,
+}
+
+/// The status of instruction execution
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum InstructionExecutionStatus {
+    /// Instruction was executed successfully
+    Success,
+
+    /// Instruction execution failed
+    /// Note: Error status is not supported yet because when CPI call fails it always terminates parent instruction
+    /// We either have to make it possible to change that behavior or add an instruction to manually set the status
+    Error,
+}
+
+/// Instruction execution flags defining how instructions are executed for a Proposal
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum InstructionExecutionFlags {
```

Comment:
Should you have an additional flag for the current behavior of `Unordered`?  Then this doesn't need to be an `Option` in the Proposal.

Reply:
Option changed to native None value in enum